src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -87,28 +86,38 @@ def publish_offers_bids(self, message):
         channel = f"{self._channel_prefix}/offers-bids/response/"
         self.myco_ext_conn.publish_json(channel, response_data)
 
-    def match_recommendations(self, message: Optional[str] = None) -> None:
+    def match_recommendations(self, message: Optional[dict] = None) -> None:
         """Receive trade recommendations and match them in the relevant market.
 
-        Matching in bulk, any pair that fails validation will cancel the operation
+        Match in bulk, any pair that fails validation will cancel the operation
         """
         if not message:
             return
 
         channel = f"{self._channel_prefix}/recommendations/response/"
-        response_data = {"event": ExternalMatcherEventsEnum.MATCH.value, "status": "success"}
+        response_data = {
+            "event": ExternalMatcherEventsEnum.MATCH.value,
+            "status": "success"}
         data = json.loads(message.get("data"))
         recommendations = data.get("recommended_matches", [])
-        try:
-            validated_recommendations = self._get_validated_recommendations(recommendations)
-            for recommendation in validated_recommendations:
+
+        response_data.update(
+            MycoExternalMatcherValidator.validate_and_report(self, recommendations))
+        if response_data["status"] != "success":
+            self.myco_ext_conn.publish_json(channel, response_data)
+            return
+        for recommendation in response_data["recommendations"]:
+            try:
+                if recommendation["status"] != "success":
+                    continue
                 market = self.markets_mapping.get(recommendation["market_id"])
+                recommendation.pop("status")
                 market.match_recommendations([recommendation])
```

Comment:
Ah okay, I couldn't do that because each recommendation can be targeted to a different market
So you can see the line `market = self.markets_mapping.get(recommendation["market_id"])`, I'm getting the market of each recommendation and posting the recommendation in this market
consider this example:
```python
recommendations = [{"market_id": "market1" ...}, {"market_id": "market2" ...}]
for recommendation in recommendations:
    market = self.markets_mapping.get(recommendation["market_id"]) # will return market1 in first iteration and market2 in the second
    market.match_recommendations([recommendation])
```